src/joypad.c

```diff
@@ -725,12 +746,41 @@ void joypad_poll(void)
     }
 
     if (check_origins) joypad_gcn_origin_check_async();
+
+    if (sys_bbplayer())
+    {
+        joypad_device_cold_t swap = joypad_devices_cold[JOYPAD_PORT_1];
+        // Respect the BBPlayer "Hack Flags" register to swap controller ports
+        switch( BB_HACK_FLAGS_SWAP_PORT1 )
+        {
+            case 1:
+                // Swap port 1 and 2
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_2];
+                joypad_devices_cold[JOYPAD_PORT_2] = swap;
+                break;
+            case 2:
+                // Swap port 1 and 3
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_3];
+                joypad_devices_cold[JOYPAD_PORT_3] = swap;
+                break;
+            case 3:
+                // Swap port 1 and 4
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_4];
+                joypad_devices_cold[JOYPAD_PORT_4] = swap;
+                break;
+        }
+    }
```

Comment:
I think it might be more efficient to change the parsing loop above. At the top of the parsing loop, we have this:

```C
        if (sys_bbplayer()) {
            // iQue has a very fixed layout for commands, and it also tends
            // to corrupt other parts of PIF-RAM. So better jump to fixed positions
            // while parsing.
            i = (port * 8) + 1;
        }
```

It would seem like changing that to:

```C
i = (bb_hack_flags_swap_port(port) * 8) + 1;
```

would be sufficient? It's already a BB-only codepath btw.